src/python/zquantum/core/circuit/_circuit.py

```diff
@@ -81,13 +81,36 @@ def n_multiqubit_gates(self):
 
         return n_mq_gates
 
+    @property
+    def symbolic_params(self):
+        """
+        Returns a list of symbolic parameters used in the circuit
+        """
+        symbolic_params = []
+        for gate in self.gates:
+            symbolic_params_per_gate = gate.symbolic_params
+            symbolic_params += symbolic_params_per_gate
+
+        return symbolic_params
```

Comment:
This doesn't give any information about gate associated with the symbol. I think that information would be useful to have, but open to hearing why that may not make sense

Reply:
If we ever need it, we can add it then.
If we never use it, the code will linger there forever ;) 
And honestly, I don't see a clear use case why I would care to know which gates are with symbolic parameters – in case of complicated ansatzes knowing that there's a parametrizable `RX` doesn't really provide me a lot of information. And which `RX` exactly is that? The first one or the 5th from the end?

Reply:
Fair point - I guess I could see it being useful, but I can't say for sure.